internal/trace/mock_span.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/tag"
+	apitrace "go.opentelemetry.io/api/trace"
+)
+
+type MockSpan struct {
+	sc     core.SpanContext
+	tracer apitrace.Tracer
+}
+
+var _ apitrace.Span = (*MockSpan)(nil)
+
+// SpancContext returns an invalid span context.
+func (ms *MockSpan) SpanContext() core.SpanContext {
+	if ms == nil {
+		core.EmptySpanContext()
+	}
+	return ms.sc
+}
+
+// IsRecordingEvents always returns false for MockSpan.
+func (ms *MockSpan) IsRecordingEvents() bool {
+	return false
+}
+
+// SetStatus does nothing.
+func (ms *MockSpan) SetStatus(status codes.Code) {
+}
+
+// SetError does nothing.
+func (ms *MockSpan) SetError(v bool) {
+}
+
+// SetAttribute does nothing.
+func (ms *MockSpan) SetAttribute(attribute core.KeyValue) {
+}
+
+// SetAttributes does nothing.
+func (ms *MockSpan) SetAttributes(attributes ...core.KeyValue) {
+}
+
+// ModifyAttribute does nothing.
+func (ms *MockSpan) ModifyAttribute(mutator tag.Mutator) {
+}
+
+// ModifyAttributes does nothing.
+func (ms *MockSpan) ModifyAttributes(mutators ...tag.Mutator) {
+}
+
+// Finish does nothing.
+func (ms *MockSpan) Finish(options ...apitrace.FinishOption) {
+}
+
+// SetName does nothing.
+func (ms *MockSpan) SetName(name string) {
+}
+
+// Tracer returns noop implementation of Tracer.
```

Comment:
That should say that it returns a mock tracer.

Reply:
fixed.